### PR TITLE
Remove lotus dependency and update current dependencies

### PR DESCRIPTION
**Pull request by xinaxu**
This PR updates underlying dependencies
- [x] Removes lotus dependency because that introduces too many unnecessary dependencies making downstream consumption very hard
- [x] Upgrade current dependency to the latest version

Breaking change - The network defaults to Testnet in the latest go-address. Consumers of this package needs to explicitly set `address.CurrentNetwork` if they would like to use for mainnet addresses.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,9 +7,7 @@ require (
 	github.com/decred/dcrd/dcrec/secp256k1/v4 v4.0.0-20210507181900-4e0be8d2fbb4
 	github.com/drand/kyber v1.1.4
 	github.com/drand/kyber-bls12381 v0.2.1
-	github.com/filecoin-project/go-address v0.0.5
-	github.com/filecoin-project/go-amt-ipld/v2 v2.1.1-0.20201006184820-924ee87a1349 // indirect
-	github.com/filecoin-project/go-state-types v0.1.1-0.20210915140513-d354ccf10379
-	github.com/filecoin-project/lotus v1.13.0
```
Comment (jsign) [inline go.mod:13]: Cool, I never liked having this dependency in this library. Looks like github.com/filecoin-project/go-state-types is rather new-ish? That's useful.

**Review by jsign — APPROVED** [review]

**Comment by jsign** [comment]
@xinaxu, the CI is failing due to some detection of CGO introduction (nice that I included this check-in CI!)
I've updated a bit [some CI stuff](https://github.com/jsign/go-filsigner/pull/9) to be sure this might not be a red herring.

Do you mind rebasing to master so we can check again?

File: go.mod (hunk 1 above)
Comment (xinaxu) [reply go.mod:13]: ty

**Comment by xinaxu** [comment]
Rebased with upstream main branch

**Comment by jsign** [comment]
> Rebased with upstream main branch

Looking good, thanks!

**Pull request merged**